### PR TITLE
change Zillion to new options dataclass

### PR DESCRIPTION
**Pull request by beauxq**
So far, I've just run unit tests.

After I test a game, I'll change from draft PR.

### CODE REVIEW

**Comment by beauxq** [comment]
Generated and played through a game with strange settings to make sure they work.
@alwaysintreble

File: worlds/zillion/options.py [hunk 1]
```diff
@@ -41,6 +41,12 @@ class VBLR(Choice):
     option_restrictive = 3
     default = 1
 
+    def to_zz_vblr(self) -> ZzVBLR:
```
Comment (alwaysintreble) [inline worlds/zillion/options.py:45]: what's the point of this? is there a bug in the options?

Comment (beauxq) [reply worlds/zillion/options.py:45]: no bug
It's replacing a `typing.cast` that was in the `validate` function below.
As I was reworking that function, I decided this was a better place to handle it.
The API of `zilliandomizer` takes a `typing.Literal`, while `Option.current_key` gives a `str`.

Comment (alwaysintreble) [reply worlds/zillion/options.py:45]: and mypy doesn't like an override that returns a different type than its parent?

Comment (beauxq) [reply worlds/zillion/options.py:45]: It does with some categories of expressions. https://mypy.readthedocs.io/en/stable/type_narrowing.html
What is not on that list is the `in` operator (used in the line `assert key in`...).
Nothing in that type narrowing list works for literals.
Here is one of the mypy issues: https://github.com/python/mypy/issues/12535

Comment (beauxq) [reply worlds/zillion/options.py:45]: After I shared that link with you, I found a way that mypy can narrow to `Literal`.
I just had to scroll down to https://mypy.readthedocs.io/en/stable/type_narrowing.html#user-defined-type-guards

**Review by alwaysintreble — APPROVED** [review]
Changes are fine. Hesitant to merge as 993 has been sitting in purgatory forever and adding more changes to it might slow it down even more and got no response last time I asked. It's probably best to just PR this branch directly against main as is and just mention it's reliant on this 🤷

**Comment by beauxq** [comment]
I understand. I wouldn't want you to pull this if it's going to slow down that PR.

**Comment by beauxq** [comment]
> got no response last time I asked

I noticed when you asked, but I guess the core devs didn't notice.

**Comment by alwaysintreble** [comment]
Screenshot_84

**Pull request merged**